src/main/java/seedu/address/logic/commands/AddFeesCommand.java

```diff
@@ -0,0 +1,139 @@
+            + ": Adds the tutorial fees to student's overdue amount\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_PAYMENT + "PAYMENT\n"
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_PAYMENT + "200";
+    public static final String MESSAGE_MARKED_PAID_SUCCESS = "Fees updated for student %1$s \n%2$s";
+
+    private static final Logger logger = LogsCenter.getLogger(AddFeesCommand.class);
+    private final Index targetIndex;
+    private final Fees feesToCharge;
+
+    /**
+     * Marks whether an existing person has paid their feesPaidByStudent for the month
+     * @param targetIndex index of the student to mark payment based on the displayed list
+     */
+    public AddFeesCommand(Index targetIndex, Fees feesToCharge) {
+        this.targetIndex = targetIndex;
+        this.feesToCharge = feesToCharge;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Person personToMarkPayment = getPersonFromModel(model);
+        Person updatedPerson = createUpdatedPerson(personToMarkPayment);
+
+        updateParticipations(model, personToMarkPayment, updatedPerson);
+
+        model.setPerson(personToMarkPayment, updatedPerson);
+        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_MARKED_PAID_SUCCESS,
+                updatedPerson.getFullName(), updatedPerson.getPayment()));
+    }
+
+    /**
+     * Retrieves the target person based on the index in the model's filtered person list.
+     */
+    private Person getPersonFromModel(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            logger.warning(String.format(Messages.MESSAGE_LOGGER_FOR_EXCEPTION, AddFeesCommand.class));
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        return lastShownList.get(targetIndex.getZeroBased());
+    }
+
+    /**
+     * Creates an updated person with the new payment balance.
+     */
+    private Person createUpdatedPerson(Person student) {
+        Payment updatedPayment = calculatePayment(student.getPayment(), feesToCharge);
+
+        return new Person(student.getName(), student.getPhone(),
+                student.getEmail(), student.getAddress(), updatedPayment, student.getTags());
+    }
+
+    /**
+     * Updates the participations of the target person in the model with the new payment status.
+     */
+    private void updateParticipations(
+            Model model, Person originalStudent, Person updatedStudent) throws CommandException {
+        List<Participation> participationsToUpdate = model.getParticipationList()
+                .filtered(participation -> participation.getStudent().isSamePerson(originalStudent));
+
```

Comment:
Should we check for whether the participations list has the corresponding students and throw an exception if it doesnt not? Cause that would mean the student isn't enrolled in a tutorial yet and it might make more sense that you add fees to a student's balance after they are enrolled in a class

Reply:
I think we should leave that level of flexibility for the admin though